application/LaunchController.cpp

```diff
@@ -215,7 +218,46 @@ void LaunchController::launchInstance()
     connect(m_launcher.get(), &LaunchTask::failed, this,  &LaunchController::onFailed);
     connect(m_launcher.get(), &LaunchTask::requestProgress, this, &LaunchController::onProgressRequested);
 
+    // Prepend Server Status
+    QStringList servers = {"authserver.mojang.com", "session.minecraft.net", "textures.minecraft.net", "api.mojang.com"};
+    QString resolved_servers = "";
+    QHostInfo host_info;
 
+    for(QString server : servers) {
+        host_info = QHostInfo::fromName(server);
+        resolved_servers = resolved_servers + server + " resolves to:\n    [";
+        if(!host_info.addresses().isEmpty()) {
+            for(QHostAddress address : host_info.addresses()) {
+                resolved_servers = resolved_servers + address.toString();
+                if(!host_info.addresses().endsWith(address)) {
+                    resolved_servers = resolved_servers + ", ";
+                }
+            }
+        } else {
+            resolved_servers = resolved_servers + "N/A";
+        }
+        resolved_servers = resolved_servers + "]\n\n";
+    }
```

Comment:
Well, if we are aiming for offline mode, querying these could lead to a very long timeouts here. basically, if we want offline, skip this.